sartography/spiff-arena
# feature/handle-empty-enum-in-jsonschema

This adds some error handling around replacing variables in enum lists in the jsonschema. If the jsonschema is using the "options_from_task_data_var" option for an anyOf and the variable is either an empty array or a string, an error message will appear telling the user what happened instead of only showing "Something went wrong".

To test:
1. Create a process model with a user task using the following react jsonshcema show below.
2. Add a prescript to user task that sets "category_enum_list" to either a string or a blank array.
3. Run the process model and a more informative error message will appear.
```
{
  "definitions": {
    "categoryEnum": {
      "title": "Category",
      "type": "string",
      "anyOf": [
        "options_from_task_data_var:category_enum_list"
      ]
    }
  },
  "properties": {
    "ndr_details": {
      "type": "object",
      "title": "Request Goods/Services",
      "description": "description",
      "properties": {
        "category": {
          "$ref": "#/definitions/categoryEnum",
          "title": "Category"
        }
      }
    }
  }
}
````
